pattern/constraint/thing/ValueConstraint.java

```diff
@@ -195,6 +197,30 @@ public FunctionalIterator<AlphaEquivalence> alphaEquals(ValueConstraint<?> that)
                 .flatMap(a -> a.alphaEqualIf(this.value.equals(that.value)));
     }
 
+    public abstract boolean isSatisfiedByValueOf(ValueConstraint<?> conclusionValueConstraint);
+
+    protected boolean evaluateEqualityPredicate(TypeQLToken.Predicate.Equality predicate, int comparatorOutput) {
+        switch (predicate){
+            case EQ:  return comparatorOutput == 0;
+            case NEQ: return comparatorOutput != 0;
+            case GT: return comparatorOutput > 0;
+            case GTE: return comparatorOutput >= 0;
+            case LT: return comparatorOutput < 0;
+            case LTE: return comparatorOutput <= 0;
+            default: throw TypeDBException.of(ILLEGAL_STATE);
+        }
+    }
+
+    protected boolean evaluateSubstringPredicate(TypeQLToken.Predicate.SubString predicate,
```

Comment:
Substring predicates use this method to evaluate the concludable pattern against the conclusion value.